src/timessquare/domain/pageparameters.py

```diff
@@ -0,0 +1,463 @@
+            return float(v)
+        except Exception as e:
+            raise PageParameterValueCastingError.for_value(v, "number") from e
+
+    def create_python_assignment(self, name: str, value: Any) -> str:
+        cast_value = self.cast_value(value)
+        return f"{name} = {cast_value}"
+
+    def create_json_value(self, value: Any) -> Any:
+        return self.cast_value(value)
+
+    def create_qs_value(self, value: Any) -> str:
+        return str(self.cast_value(value))
+
+
+class BooleanParameterSchema(PageParameterSchema):
+    """A boolean-type parameter schema."""
+
+    def cast_value(self, v: Any) -> bool:
+        """Cast a value to its Python type."""
+        if isinstance(v, str):
+            if v.lower() == "true":
+                return True
+            elif v.lower() == "false":
+                return False
+            else:
+                raise PageParameterValueCastingError.for_value(v, "boolean")
+        elif isinstance(v, bool):
+            return v
+        raise PageParameterValueCastingError.for_value(v, "boolean")
+
+    def create_python_assignment(self, name: str, value: Any) -> str:
+        cast_value = self.cast_value(value)
+        return f"{name} = {cast_value}"
+
+    def create_json_value(self, value: Any) -> Any:
+        return self.cast_value(value)
+
+    def create_qs_value(self, value: Any) -> str:
+        # The query string follows the JSON format, so lowercase true/false.
+        return str(self.cast_value(value)).lower()
+
+
+class DateParameterSchema(PageParameterSchema):
+    """A date-type parameter schema."""
+
+    def cast_value(self, v: Any) -> date:
+        """Cast a value to its Python type."""
+        try:
+            # Parse an ISO 8601 date string
+            if isinstance(v, str):
+                return date.fromisoformat(v)
+            elif isinstance(v, date):
+                return v
+        except Exception as e:
+            raise PageParameterValueCastingError.for_value(v, "date") from e
+
+        raise PageParameterValueCastingError.for_value(v, "date")
+
+    def create_python_imports(self) -> list[str]:
+        return ["import datetime"]
+
+    def create_python_assignment(self, name: str, value: Any) -> str:
+        date_value = self.cast_value(value)
+        str_value = date_value.isoformat()
+        return f'{name} = datetime.date.fromisoformat("{str_value}")'
+
+    def create_json_value(self, value: Any) -> Any:
+        return self.cast_value(value).isoformat()
+
+    def create_qs_value(self, value: Any) -> str:
+        return self.cast_value(value).isoformat()
+
+
+class DatetimeParameterSchema(PageParameterSchema):
+    """A datetime-type parameter schema that is timezone-aware."""
+
+    def cast_value(self, v: Any) -> datetime:
+        """Cast a value to its Python type."""
+        date_value = self._cast_to_datetime(v)
+        if date_value.tzinfo is None:
+            # Force the timezone to UTC if naieve
```

Comment:
```suggestion
            # Force the timezone to UTC if naive
```

Reply:
What do you think about requiring timezones to be specified instead of assuming UTC?